### PR TITLE
Include incubation feature in setup

### PR DESCRIPTION
**Pull request by merks**
Use DisplayRealm.getReal rather than SWTObservables.getRealm. Accommodate generates in generics in GEF.

https://github.com/eclipse/nebula/issues/527

### CODE REVIEW

**Comment by merks** [comment]
@lcaron 

The build are failing like for this PR for me locally as well. I think it needs version changes like this:

image

What I hope to get working is a local launch configuration for the Tycho build like this:

image

If I get that working, I can better determine which projects are build built and should be in the workspace and then eventually automating the publishing of update sites like I've done for other projects, most recently for GEF Classic...

File: widgets/datechooser/org.eclipse.nebula.widgets.datechooser/src/org/eclipse/nebula/widgets/datechooser/DateChooserObservableValue.java [hunk 1]
```diff
@@ -63,7 +63,7 @@ public void widgetDisposed(DisposeEvent e) {
 	};
 
 	public DateChooserObservableValue(final DateChooser dateChooser, int updateEventType) {
-		this(SWTObservables.getRealm(dateChooser.getDisplay()), dateChooser, updateEventType);
+		this(DisplayRealm.getRealm(dateChooser.getDisplay()), dateChooser, updateEventType);
```
Comment (laeubi) [inline widgets/datechooser/org.eclipse.nebula.widgets.datechooser/src/org/eclipse/nebula/widgets/datechooser/DateChooserObservableValue.java]:
```suggestion
		this(Realm.getCurrent(), dateChooser, updateEventType);
```
at least that's what is used in the DatabindingFramework internally when no realm is passed.

Comment (merks) [reply widgets/datechooser/org.eclipse.nebula.widgets.datechooser/src/org/eclipse/nebula/widgets/datechooser/DateChooserObservableValue.java]: Did you mean Realm.getDefault()?  I don't claim to understand the subtleties of this, so choose to replace it with something that appears to be 100% semantically equivalent with what was there before.  E.g., it looks like getDefault works only on the display thread, but DisplayRealm.getRealm works on any thread, I think.  Of course, I don't know whether that is relevant here either...

Comment (laeubi) [reply widgets/datechooser/org.eclipse.nebula.widgets.datechooser/src/org/eclipse/nebula/widgets/datechooser/DateChooserObservableValue.java]: Yes `Realm.getDefault()` if you search for references you will se that this is always used when no explicit realm is passed in e.g. you can even look at the AbstractObservableValue default constructor that is extended here and I think its better to stick to this convention than trying to be "smart" here, `Realm.getDefault()` works on any thread that is currently a realm thread and one should only create observables from such thread or pass an explicit realm.

Comment (merks) [reply widgets/datechooser/org.eclipse.nebula.widgets.datechooser/src/org/eclipse/nebula/widgets/datechooser/DateChooserObservableValue.java]: I see what you mean, e.g., 

image

Comment (merks) [reply widgets/datechooser/org.eclipse.nebula.widgets.datechooser/src/org/eclipse/nebula/widgets/datechooser/DateChooserObservableValue.java]: There was another place using that "pattern" so I changed it too.

**Comment by lcaron** [comment]
@merks I've fixed the build. Can you please rebase and keep me informed ?

**Comment by merks** [comment]
FYI, I'm out of office until Friday.  Hopefully I'll have time then.

**Comment by merks** [comment]
I think I have some mistaken conceptions about what should be in the target platform...  I can't look at this further until I'm back.

**Comment by merks** [comment]
Okay, last try for today.  I was able to do both builds locally now...

**Comment by merks** [comment]
@lcaron  @wimjongman 

This can be merged and then I can look again at how we might improve the production of update sites...

**Comment by wimjongman** [comment]
Wonderful, thanks Ed.

**Comment by wimjongman** [comment]
Is it easier for you if we make you a committer?

**Comment by merks** [comment]
Yes, that would allow me to prototype everything without bothering you folks nearly as much...

**Comment by lcaron** [comment]
Thank you Ed. You deserve to be a commiter. I'm gonna submit the request.

**Pull request merged**